### ISSUE TITLE
PersistenceQuery stream is not notified of mongo connection exceptions.

### ISSUE THREAD

**Issue by PiotrFras** [scullxbones/akka-persistence-mongo#155]
Mongo: 3.2,
Query: 2.4.18

Given consumer and producer:

```
object Consumer {

  def main(args: Array[String]): Unit = {

    val config = ConfigFactory.parseString(
      s"""
         |
         |akka.persistence.journal.plugin = "akka-contrib-mongodb-persistence-journal"
         |akka.remote {
         |    enabled-transports = ["akka.remote.netty.tcp"]
         |    netty.tcp {
         |      port = 2552
         |      bind-hostname = "0.0.0.0"
         |    }
         |  }
         |
       """.stripMargin).withFallback(ConfigFactory.load())

    implicit val system = ActorSystem("Consumer", config)

    implicit val mat = ActorMaterializer(ActorMaterializerSettings(system).withSupervisionStrategy { _ =>
      println("restarting the stream")
      Supervision.Restart
    })

    val query = PersistenceQuery(system).readJournalFor[ScalaDslMongoReadJournal](MongoReadJournal.Identifier)
    query.eventsByPersistenceId("test", 0, Long.MaxValue).map(_.toString)
      .recover { case thrown => s"Recovered from:  ${thrown.getMessage}" }
      .runForeach(event => println(s"Got event: [$event]"))
  }

}
```

```
object Producer {

  def main(args: Array[String]): Unit = {

    val config = ConfigFactory.parseString(
      s"""
         |
         |akka.persistence.journal.plugin = "akka-contrib-mongodb-persistence-journal"
         |
       """.stripMargin).withFallback(ConfigFactory.load())

    implicit val system = ActorSystem("Producer", config)

    implicit val mat = ActorMaterializer()

    val actor = system.actorOf(Props(new Actor))

    Source.tick(1.second, 1.second, "OBJECT").to(Sink.actorRef(actor, "done")).run()
  }

  class Actor extends PersistentActor {

    override def receiveRecover: Receive = {
      case _ =>
    }

    override def receiveCommand: Receive = {
      case "done" => self ! PoisonPill
      case o => persist(o) { _ => }) match {
        case Success(_) => println("event persisted")
        case Failure(_) => println("event failed")
      }
    }

    override def persistenceId: String = "test"
  }

}
```

consumer stream `eventsByPersistenceId` is not notified of any mongo connection exceptions (mongo is restarted or down), which results in blocked/empty stream. Supervision strategy is not triggered.

Consumer log:

```
Got event: [EventEnvelope(1498140250796,test,1,OBJECT)]
Got event: [EventEnvelope(1498140251332,test,2,OBJECT)]
Got event: [EventEnvelope(1498140252334,test,3,OBJECT)]
Got event: [EventEnvelope(1498140253328,test,4,OBJECT)]
Got event: [EventEnvelope(1498140254331,test,5,OBJECT)]
Got event: [EventEnvelope(1498140255337,test,6,OBJECT)]
Got event: [EventEnvelope(1498140256330,test,7,OBJECT)]
Got event: [EventEnvelope(1498140257336,test,8,OBJECT)]
[WARN] [2017-06-22 15:04:17.938] [org.mongodb.driver.connection] Got socket exception on connection [connectionId{localValue:3, serverValue:11}] to 127.0.0.1:27017. All connections to 127.0.0.1:27017 will be closed.
```

To reproduce:

0. Start Mongodb.
1. Start consumer.
2. Start producer.
3. Observe logs.
4. Stop mongo.
5. Observe exception in Consumer console.
6. Restart mongo.
7. Restart producer.
8. Consumer stream is not resumed.

**Comment by scullxbones** [comment]
Hi @PiotrFras - thanks for the report.

This behavior doesn't surprise me given the way that the live update queries work, using the event bus.  I'll have to think about how the error condition gets transmitted to the query's `Source`.

**Comment by PiotrFras** [comment]
Hi @scullxbones,

I managed to fix it. I could submit PR however tests are not stable and keep failing. Would you like to still have a look at the PR?

**Comment by scullxbones** [comment]
Sure - i'll happily take a look, thanks!

**Comment by bfil** [comment]
We have some pending changes in a PR against our fork here:

https://github.com/ovotech/akka-persistence-mongo/pull/1

I think it still needs tweaking, I'm pretty sure it's better to stop the actor rather then changing the behaviour when the initial cursor cannot be created here:
https://github.com/ovotech/akka-persistence-mongo/pull/1/files#diff-71d98b13abcfefba0e10ffbae5becdddR342

Some thoughts from your side would be nice though :)

**Comment by scullxbones** [comment]
I think that's the right behavior, `onErrorThenStop` if the cursor can't get created.

**Comment by scullxbones** [comment]
Hi @PiotrFras / @bfil  - wanted to check in and see how this is going, as I haven't seen any updates in ~ 1 month.

I may have some spare time coming up and can jump in and help out.

**Comment by scullxbones** [comment]
Fixed by #199
